xmake/plugins/lua/xmake.lua

```diff
@@ -56,8 +61,33 @@ task("lua")
                 import("scripts." .. name).main(unpack(option.get("arguments") or {}))
             end
         else
+            local replhistory
+            if enable_readline then
+                -- clear history
+                readline.clear_history()
+
+                -- load history
+                replhistory = history.load("replhistory") or {}
+                for _, ln in ipairs(replhistory) do
+                    readline.add_history(ln)
+                end
+            end
+
             -- enter interactive mode
             sandbox.interactive()
+
```

Comment:
There are some reason for it
* I wanna reuse `history` API to store replhistory
* Duty should be clear. `sandbox.interactive()` should only make sandbox interactive but not history store. History storing is the job of upper level
* I have considered what you think. So I put readline into `lib.readline` but not in `core` as an optional library. In this way it's clear and give enough info that this is **optional** and not available always. That's why I open a new namespace `lib`